com.unity.netcode.gameobjects/Runtime/Utility/FixedUnmanagedArray.cs

```diff
@@ -0,0 +1,181 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// For each usage of FixedUnmanagedArray, a storage class needs to be created for it.
+    /// Rather than providing a huge list of predefined storage classes, each use should just
+    /// create their own at the correct size (in bytes). The size should be an even multiple
+    /// of the value size being stored.
+    ///
+    /// Example:
+    ///
+    /// [StructLayout(LayoutKind.Explicit, Size = 256 * sizeof(int))]
+    /// struct FixedStorageInt256 : IFixedArrayStorage
+    /// {
+    /// }
+    /// </summary>
+    public interface IFixedArrayStorage
+    {
+
+    }
+
+    public struct FixedUnmanagedArray<TPropertyType, TStorageType> : IReadOnlyList<TPropertyType>
```

Comment:
I saw very few places using this, maybe we don't need to introduce it today and try to use `NativeArray` whenever possible? 👀

Reply:
Yeah... I expected to need to use it more... I can just change the one and only usage to just do a string.

Reply:
Removed.